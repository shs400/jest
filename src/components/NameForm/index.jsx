import React, { Component } from 'react';
import PropTypes from 'prop-types';

class NameForm extends Component {
  static propTypes = {
    onInsert: PropTypes.func,
  }

  static defaultProps = {
    onInsert: () => {},
  }

  constructor(props) {
    super(props);
    this.state = {
      name: '',
    };
  }

  onChange = (e) => {
    const { value } = e.target
    this.setState({
      name: value,
    });
  }

  onSubmit = (e) => {
    e.preventDefault();
    const { name } = this.state;
    const { onInsert } = this.props;
    // 이름을 추가하고, name 값 초기화
    if (!name) return false;
    onInsert(name);
    return this.setState({
      name: '',
    });
  }

  render() {
    const { name } = this.state;
    return (
      <form onSubmit={this.onSubmit}>
        <label htmlFor="name">
          이름
          <input type="text" id="name" value={name} onChange={this.onChange} />
        </label>
        <button type="submit">등록</button>
      </form>
    );
  }
}

export default NameForm;
