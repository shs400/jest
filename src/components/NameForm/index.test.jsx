import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import NameForm from './index';
configure({ adapter: new Adapter() });

describe('NameFrom', () => {
  let component = null;

  it('renders correctly', () => {
    component = shallow(<NameForm />);
  });

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  describe('insert new text', () => {
    it('has a form', () => {
      expect(component.find('form').exists()).toBe(true);
      // form이 있는지 확인
    });

    it('has an input', () => {
      expect(component.find('input').exists()).toBe(true);
      // input이 있는지 확인
    });

    it('simulates input change', () => {
      const mockEvent = {
        target: {
          value: 'hello'
        }
      }

      component.find('input').simulate('change', mockEvent);
      // 시뮬레이트 합니다. 두번째 파라미터는 이벤트 객체
      expect(component.state().name).toBe('hello');
    });

    it('simulates input submit', () => {
      const mockEvent = {
        preventDefault: () => null
      };
      component.find('form').simulate('submit', mockEvent);
      expect(component.state().name).toBe('');
    });

  });
});