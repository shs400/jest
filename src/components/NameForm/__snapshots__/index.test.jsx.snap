// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameFrom matches snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    htmlFor="name"
  >
    이름
    <input
      id="name"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <button
    type="submit"
  >
    등록
  </button>
</form>
`;
